stdlib/jdbc/src/main/java/org/ballerinax/jdbc/datasource/SQLDatasource.java

```diff
@@ -61,7 +63,9 @@ public SQLDatasource init(SQLDatasourceParams sqlDatasourceParams) {
             throw ErrorGenerator.getSQLDatabaseError(e);
         }
         try (Connection con = getSQLConnection()) {
-            databaseProductName = con.getMetaData().getDatabaseProductName().toLowerCase(Locale.ENGLISH);
+            DatabaseMetaData metaData = con.getMetaData();
+            databaseProductName = metaData.getDatabaseProductName().toLowerCase(Locale.ENGLISH);
+            this.supportsGetGeneratedKeys = metaData.supportsGetGeneratedKeys();
```

Comment:
We cannot rely on this API to check whether DB driver supports `get generated keys`. Because there is no guarantee that all DB driver implements this function. IMO we should not use this for our internal logic